src/test/HostActivationTests/FrameworkResolution/FrameworkResolutionCommandResultExtensions.cs

```diff
@@ -20,12 +20,17 @@ public static AndConstraint<CommandResultAssertions> DidNotFindCompatibleFramewo
 
         public static AndConstraint<CommandResultAssertions> FailedToSoftRollForward(this CommandResultAssertions assertion, string frameworkName, string newVersion, string previousVersion)
         {
-            return assertion.HaveStdErrMatching($".*The specified framework '{frameworkName}', version '{newVersion}', patch_roll_fwd=[0-1], roll_fwd_on_no_candidate_fx=[0-2] cannot roll-forward to the previously referenced version '{previousVersion}'.*");
+            return assertion.HaveStdErrMatching($".*The specified framework '{frameworkName}', version '{newVersion}', apply_patches=[0-1], roll_forward=[0-2] cannot roll-forward to the previously referenced version '{previousVersion}'.*");
```

Comment:
Is this a fatal error? Or does it signify a retry?  The assertion `FailedToSoftRollForward` seems to imply a non-recoverable failure.  Maybe this should be `HasWarningFailedToSoftRollForward`

Reply:
It is a fatal error, but I'm not aware of a good way to add the `Should().Fail()` to the constraints (as that statement changes the return type). I'll take a look if it can be somehow made possible.

Reply:
This should now be mostly resolved with the usage of fluent assertion on all the tests.